pkg/webhooks/mutator/virt-launcher-mutator.go

```diff
@@ -0,0 +1,234 @@
+	"k8s.io/apimachinery/pkg/api/equality"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	"github.com/kubevirt/hyperconverged-cluster-operator/api/v1beta1"
+)
+
+var _ admission.Handler = &VirtLauncherMutator{}
+
+const (
+	cpuLimitToRequestRatioAnnotation    = "kubevirt.io/cpu-limit-to-request-ratio"
+	memoryLimitToRequestRatioAnnotation = "kubevirt.io/memory-limit-to-request-ratio"
+
+	launcherMutatorStr = "virtLauncherMutator"
+)
+
+type VirtLauncherMutator struct {
+	cli          client.Client
+	hcoNamespace string
+	decoder      *admission.Decoder
+	logger       logr.Logger
+}
+
+func NewVirtLauncherMutator(cli client.Client, hcoNamespace string) *VirtLauncherMutator {
+	return &VirtLauncherMutator{
+		cli:          cli,
+		hcoNamespace: hcoNamespace,
+		logger:       log.Log.WithName("virt-launcher mutator"),
+	}
+}
+
+func (m *VirtLauncherMutator) Handle(ctx context.Context, req admission.Request) admission.Response {
+	m.logInfo("Starting virt-launcher mutator handling")
+
+	if req.Operation != admissionv1.Create {
+		m.logInfo("not a pod creation - ignoring")
+		return admission.Allowed(ignoreOperationMessage)
+	}
+
+	launcherPod := &k8sv1.Pod{}
+	err := m.decoder.Decode(req, launcherPod)
+	if err != nil {
+		m.logErr(err, "cannot decode virt-launcher pod")
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+	originalPod := launcherPod.DeepCopy()
+
+	hco, err := getHcoObject(ctx, m.cli, m.hcoNamespace)
+	if err != nil {
+		m.logErr(err, "cannot get the HyperConverged object")
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	enforceCpuLimits, enforceMemoryLimits, err := m.getResourcesToEnforce(ctx, launcherPod.Namespace, hco)
+	if err != nil {
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	if !enforceCpuLimits && !enforceMemoryLimits {
+		return admission.Allowed(ignoreOperationMessage)
+	}
+
+	if err := m.handleVirtLauncherCreation(launcherPod, hco, enforceCpuLimits, enforceMemoryLimits); err != nil {
+		m.logErr(err, "failed handling launcher pod %s", launcherPod.Name)
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	allowResponse := m.getAllowedResponseWithPatches(launcherPod, originalPod)
+	m.logInfo("mutation completed successfully for pod %s", launcherPod.Name)
+	return allowResponse
+}
+
+func (m *VirtLauncherMutator) handleVirtLauncherCreation(launcherPod *k8sv1.Pod, hco *v1beta1.HyperConverged, enforceCpuLimits, enforceMemoryLimits bool) error {
+	var cpuRatioStr, memRatioStr string
+	if enforceCpuLimits {
+		cpuRatioStr = hco.Annotations[cpuLimitToRequestRatioAnnotation]
+	}
+	if enforceMemoryLimits {
+		memRatioStr = hco.Annotations[memoryLimitToRequestRatioAnnotation]
+	}
+
+	if enforceCpuLimits {
+		err := m.setResourceRatio(launcherPod, cpuRatioStr, cpuLimitToRequestRatioAnnotation, k8sv1.ResourceCPU)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
why this block is not at the same block at lines 90-91 above?
same comment for the other block of:
`if enforceMemoryLimits {`